### PR TITLE
Add notes to choice editor

### LINKED ISSUE #67: Add secret note to choice editor
From user feedback:

> The only thing I would add, is the possibility for the writer to have a note section for each scene block, that will not be visible to the reader. It would help to plan and take notes of scenes we might later want to improve or expand, etc. Just a new text field at the bottom of the right menu when editing a block, to take a "secret" note.

So a new text box below the Choices selection:
Screen Shot 2020-01-13 at 4 12 02 PM

Longterm I think it'd be nice to add a settings panel that would let you toggle on/off features like this. But for now adding all the things is probably fine.

### PR DESCRIPTION
**Pull request by DanielSBrown**
Fixes #67 

This PR adds an Editor Notes textarea to the Editor view of a story. Instead of directly copying the Content textarea code, we decided to abstract it out to function component `SceneEditorTextAreaField`.

While developing, we found an issue with saving certain fields on the editor due to an idempotence issue with how previous versions were being compared to figure out if the save actually needed to fire. This PR addresses that issue as well.

### CODE REVIEW

File: app/javascript/src/components/SceneEditor.tsx [hunk 1]
```diff
@@ -44,37 +40,46 @@ class SceneEditor extends React.Component<SceneEditorProps> {
...
+          instructionalText=""
+          onChange={this.onChangeContent.bind(this)}
 
+        />
         <div className="SceneEditorField">
           <h3 className="SceneEditorHeading">Choices</h3>
 
           <ChoiceEditor focus={focus} requestPaint={requestPaint} />
-        </div>
+          </div>
```
Comment (efatsi) [inline app/javascript/src/components/SceneEditor.tsx]: bad spacing here

File: app/javascript/src/components/SceneEditor.tsx [hunk 2]
```diff
@@ -44,37 +40,46 @@ class SceneEditor extends React.Component<SceneEditorProps> {
...
-        buttons: ['format', 'bold', 'italic', 'lists'],
-        callbacks: {
-          synced: (html: string) => this.onChange(html)
-        }
-      })
-    }
-  }
 
-  private onChange(html: string) {
+   private onChangeContent(html: string) {
```
Comment (efatsi) [inline app/javascript/src/components/SceneEditor.tsx]: extra space character here as well

File: app/javascript/src/components/SceneEditor.tsx [hunk 3]
```diff
@@ -120,3 +125,40 @@ export default ({ focus, requestPaint, onClear }: ConsumerProps) => {
...
+  React.useEffect(() => {
+    if (inputRef.current) {
+      $R(inputRef.current, {
+        buttons: ['format', 'bold', 'italic', 'lists'],
+        callbacks: {
+          synced: (html: string) => onChange(html)
+        }
+      })
+    }
+  }, [inputRef.current])
```
Comment (efatsi) [inline app/javascript/src/components/SceneEditor.tsx]: neat!

File: app/javascript/src/components/Editor.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ interface EditorProps {
 class Editor extends React.Component<EditorProps, EditorState> {
   engine: DiagramEngine
   model: DiagramModel
-  lastSavedState: ApplicationState
+  lastSavedState: string
```
Comment (joeyjoejoejr) [inline app/javascript/src/components/Editor.tsx]: Can you explain why this change is necessary?  The issue I have with this, is that the string type is weaker than the ApplicationState type.  Now pretty much any object could be serialized into a string, and cause type errors down the line.

File: app/javascript/src/components/SceneEditor.tsx (hunk 2 above)
Comment (DanielSBrown) [reply app/javascript/src/components/SceneEditor.tsx]: this and above issue resolved in fa8add8

File: app/javascript/src/components/SceneEditor.tsx [hunk 5]
```diff
@@ -44,37 +40,46 @@ class SceneEditor extends React.Component<SceneEditorProps> {
...
-          <label className="SceneEditorHeading" htmlFor="content">Content</label>
-          <textarea name="content" ref={this.editor} defaultValue={text} />
-        </div>
+        <SceneEditorTextAreaField
+          name="content"
+          title="Content"
+          defaultValue={text}
+          placeholderText=""
+          instructionalText=""
+          onChange={this.onChangeContent.bind(this)}
```
Comment (joeyjoejoejr) [inline app/javascript/src/components/SceneEditor.tsx]: The style in this app so far is not to bind in render, but to define these functions a class propertires so onChangeContent should be defined like:

```
onChangeContent = (html: string) => {
// code here
}
```

File: app/javascript/src/components/Editor.tsx (hunk 4 above)
Comment (DanielSBrown) [reply app/javascript/src/components/Editor.tsx]: This came up in a discussion with @efatsi. When it was an application state, it was causing problems with line 325, since `lastSavedState` would sometimes update at the same time as `newState`. He could give a little more insight into exactly why, since he had worked on a potential fix in 0899f7ea. He mentioned the above approach would work, since a string for `lastSavedState` would not cause that strange problem of `lastSavedState` updating when `newState` was updated, causing line 326 of the old version to report that there was no change when there was, resulting in a save not happening where it should. 

I understand from your explanation why this may cause errors down the line; do you think we should look into Eli's commit to try to get something like that to work?

Comment (efatsi) [reply app/javascript/src/components/Editor.tsx]: Yeah, portions of the app don't work idempotently, so updates to the current state pollute the `lastSavedState`. My commit should do the trick if we want to keep `lastSavedState` from being a string (or at least a subset of it that dealt with this). It writes a clean dup of the state to `lastSavedStaet`. It's sitting in a branch still so that's why this is still a problem on master.

That said, I'm not all that worried about the `lastSavedState` being a string since as it stands now it's just being used for this one purpose (and in this purpose we turn it into a string for the equality check). But I don't have a brain for Typescript yet so differ to others for best practices here.

File: app/javascript/src/components/SceneEditor.tsx (hunk 5 above)
Comment (DanielSBrown) [reply app/javascript/src/components/SceneEditor.tsx]: Addressed in 97241e4
